OSDOCS-6393: revise install instructions embedding in image (#61003)
openshift/openshift-docs: modules/microshift-adding-repos-to-image-builder.adoc

.Prerequisites
* Your build host meets the Image Builder system requirements.
* You have installed and set up Image Builder and the `composer-cli` tool.

> * You have root-user access to your build host.

Reviewer: Would it be better to use the term superuser here ?

Author: Hmm, good question. I don't see anything in the IBM Style Guide about it, nor do we use it often in the OCP Docs. I'll see if someone can point me to guidance on docs use of the term. TBD

Author: @dhensel-rh We _can_ use it, but haven't. I do not think it is worth a sweep through the docs to update this term consistently, so I'll leave it. The note we have in our supplementary style guide is that the term is more common in Solaris documentation than Linux.

.Procedure

> . Enable the {product-title} RPM repositories on the build host by running the following command:

Reviewer 2: This is a wrong comment. Please, check my original script comment for a better description

+
[source,terminal]
----

> # cat > rhocp-4.13.toml <<EOF

Reviewer 2: ```suggestion
$ cat > rhocp-4.13.toml <<EOF
```

> $ sudo cat > rhocp-4.13.toml <<EOF

Reviewer 3: What directory should that file end up in?

I suspect using `sudo cat` is not going to do the right thing because of the way the shell interprets the redirection. We've used `sudo tee` in other places to get around that. On the other hand, it doesn't look like `sudo composer-cli sources add` below would require the file in this step be created by root, so maybe we don't need the sudo here at all?

Author: It was `# cat`, but the rest of the commands are `$ sudo`, so I wanted them to be consistent. This will come up in peer review, so let's figure out which way it needs to be. We have root access as a prereq for the entire procedure, so whichever command syntax is correct is fine.

Author: @ggiguash Do we have a directory to list for the end user? Or should that be specified somewhere in this command?

Reviewer 3: I was able to run the `cat` step as a regular user and then use the file as root. It doesn't really matter where the file is, the command that loads it can read the file and it passes data to the service so the service does not need the file itself.

Author: @dhellmann So we don't need to answer your original question: "What directory should that file end up in?"

Reviewer 2: Yes, it's not necessary to specify the file location. They are created in the current directory.

id = "rhocp-4.13"
name = "Red Hat OpenShift Container Platform 4.13 for RHEL 9"
type = "yum-baseurl"
url = "https://cdn.redhat.com/content/dist/layered/rhel9/$(uname -m)/rhocp/4.13/os"
check_gpg = true

> check_ssl = false

Reviewer 3: ```suggestion
check_ssl = true
```

Reviewer 3: We should not disable SSL verification on the repos. Please test this step again if that's necessary.

Reviewer: I will check the steps again with this change

Reviewer: I've checked the check_ssl = true change and it looks ok

system = false
rhsm = true
EOF
----

> . Create an Image Builder configuration file for adding the needed RPM repository by running the following command:

Reviewer 2: ```suggestion
. Create an Image Builder configuration file for adding the `fast-datapath` RPM repository by running the following command:
```

+
[source,terminal]
----

> # cat > fast-datapath.toml <<EOF

Reviewer 2: ```suggestion
$ cat > fast-datapath.toml <<EOF
```

id = "fast-datapath"
name = "Fast Datapath for RHEL 9"
type = "yum-baseurl"
url = "https://cdn.redhat.com/content/dist/layered/rhel9/$(uname -m)/fast-datapath/os"
check_gpg = true

> check_ssl = false

Reviewer 3: ```suggestion
check_ssl = true
```

Author: @dhellmann With @dhensel-rh 's lgtm, I think these have been QE'd as-is. Could you confirm, @dhensel-rh?

Reviewer: I will check the steps again with this change

Reviewer: I've checked the _check_ssl = true_ change and it looks ok

* Your build host meets the Image Builder system requirements.
* You have installed and set up Image Builder and the `composer-cli` tool.
* You have root-user access to your build host.

.Procedure

. Enable the {product-title} RPM repositories on the build host by running the following command:
+
[source,terminal]
----
# cat > rhocp-4.13.toml <<EOF
id = "rhocp-4.13"
name = "Red Hat OpenShift Container Platform 4.13 for RHEL 9"
type = "yum-baseurl"
url = "https://cdn.redhat.com/content/dist/layered/rhel9/$(uname -m)/rhocp/4.13/os"
check_gpg = true
check_ssl = false
system = false
rhsm = true
EOF
----

. Create an Image Builder configuration file for adding the needed RPM repository by running the following command:
+
[source,terminal]
----
# cat > fast-datapath.toml <<EOF
id = "fast-datapath"
name = "Fast Datapath for RHEL 9"
type = "yum-baseurl"
url = "https://cdn.redhat.com/content/dist/layered/rhel9/$(uname -m)/fast-datapath/os"
check_gpg = true
check_ssl = false
system = false
rhsm = true
EOF
----

. Add the sources to the Image Builder by running the following commands:
+
[source,terminal]
----
# sudo composer-cli sources add rhocp-4.13.toml
----
+
[source,terminal]
----

> # sudo composer-cli sources add fast-datapath.toml

Reviewer 2: ```suggestion
sudo composer-cli sources add fast-datapath.toml
```
